handlers/src/lib.rs

```diff
@@ -1,100 +1,70 @@
-        let reference = format!("refs/{}", query_params.reference);
-        //TODO return proper content type depending on the content of the blob
-        self.git_ops
-            .cat_file(&repo_path, &reference, &query_params.file)
-            .map(Binary::from)
-            .expect("Can't cat file")
-    }
-}
+pub struct CatFileResponse(pub Result<Vec<u8>, String>);
 
-impl RepoHandler {
-    pub fn new(repo_root: String) -> RepoHandler {
-        RepoHandler {
-            repo_root,
-            git_ops: Box::new(LibGitOps {}),
+impl<A, M> MessageResponse<A, M> for CatFileResponse
+where
+    A: Actor,
+    M: Message<Result = CatFileResponse>,
+{
+    fn handle<R: ResponseChannel<M>>(self, _: &mut A::Context, tx: Option<R>) {
+        if let Some(tx) = tx {
+            tx.send(self);
         }
     }
 }
 
-#[cfg(test)]
-mod tests {
-
-    use super::RepoHandler;
-    use actix_web::http::StatusCode;
-    use actix_web::test;
-    use actix_web::{Binary, Body};
-    use git::GitOps;
-    use std::path;
-
-    struct TestGitOps {
-        res: Vec<u8>,
-    }
+pub struct GitRepos {
+    repos: HashMap<String, Repository>,
+    ops: Box<GitOps>,
+}
 
-    impl GitOps for TestGitOps {
-        fn cat_file(
-            &self,
-            _repo_path: &path::Path,
-            _reference: &str,
-            _filename: &str,
-        ) -> Result<Vec<u8>, git2::Error> {
-            Ok(self.res.to_owned())
-        }
-    }
+impl Actor for GitRepos {
+    type Context = Context<Self>;
+}
 
-    fn bin_ref(body: &Body) -> &Binary {
-        match *body {
-            Body::Binary(ref bin) => bin,
-            _ => panic!(),
+impl GitRepos {
+    pub fn new(repos: HashMap<String, Repository>) -> GitRepos {
+        GitRepos {
+            repos,
+            ops: Box::new(LibGitOps {}),
         }
     }
+}
 
-    #[test]
-    fn it_returns_the_content_of_the_file_by_cat_file() {
-        let rp = RepoHandler {
-            repo_root: "not-used".to_string(),
-            git_ops: Box::new(TestGitOps {
-                res: b"hello".to_vec(),
-            }),
-        };
+impl Handler<CatFile> for GitRepos {
+    type Result = CatFileResponse;
 
-        let resp = test::TestRequest::with_header("content-type", "application/json")
-            .param("repo", "client-config.git")
-            .uri("/repo/?reference=the-reference&file=the-file")
-            .run(&rp)
-            .expect("can't run test request");
-        assert_eq!(resp.status(), StatusCode::OK);
-        assert_eq!(bin_ref(resp.body()), &Binary::from_slice(b"hello"));
+    fn handle(&mut self, task: CatFile, _: &mut Self::Context) -> Self::Result {
+        CatFileResponse(match self.repos.get(&task.repo_key) {
+            Some(repo) => self
+                .ops
+                .cat_file(repo, &task.reference, &task.filename)
+                .map_err(|x| x.to_string()),
+            None => Err("No repo found".to_string()),
```

Comment:
Probably need something more like?

```suggestion
            None => Err(format!("No repo found with name {}", &task.repo_key)),
```